compendium/LogServiceImpl/src/LogServiceImpl.cpp

```diff
@@ -6,110 +6,192 @@
 #include "cppmicroservices/ServiceReference.h"
 
 #include "LogServiceImpl.hpp"
+#include "LoggerFactoryImpl.hpp"
+#include "LoggerImpl.hpp"
 
-namespace cppmicroservices
-{
-    namespace logservice
-    {
-        std::string
-        GetExceptionMessage(std::exception_ptr const& ex)
+namespace cppmicroservices::logservice
+{     
+        LogServiceImpl::LogServiceImpl(std::string const& loggerName)
         {
-            std::string message = "\nException logged: ";
-            if (ex)
-            {
-                std::ostringstream stream;
-                try
-                {
-                    std::rethrow_exception(ex);
-                }
-                catch (std::exception const& e)
-                {
-                    message += std::string(typeid(e).name()) + " : " + e.what();
-                }
-            }
-            else
-            {
-                message += "none";
-            }
-
-            return message;
-        }
-
-        std::string
-        GetServiceReferenceInfo(ServiceReferenceBase const& sr)
+            logger = getLogger(loggerName);    
+	}
+   
+        void
+        LogServiceImpl::Log(SeverityLevel level, std::string const& message)
         {
-            std::ostringstream stream;
-            stream << "\nServiceReference: " << sr;
-            return stream.str();
-        }
 
-        LogServiceImpl::LogServiceImpl(std::string const& loggerName)
-        {
-            auto sink = std::make_shared<spdlog::sinks::stdout_color_sink_mt>();
-            m_Logger = std::make_shared<spdlog::logger>(loggerName, std::move(sink));
-            m_Logger->set_pattern("[%T] [%P:%t] %n (%^%l%$): %v");
-            m_Logger->set_level(spdlog::level::trace);
-        }
+           auto currLogger = std::atomic_load(&logger);
+           if (!currLogger)
+           {
+               return;
+           }
+
+           switch (level)
+           {
+               case SeverityLevel::LOG_DEBUG:
+               {
+		   currLogger->debug(message);
+	           break;
+	       }
+	       case SeverityLevel::LOG_INFO:
+	       {
```

Comment:
I believe you should be able to remove the '{' and '}' around each case, i.e.:

               case SeverityLevel::LOG_DEBUG:
		   currLogger->debug(message);
	           break;
	       case SeverityLevel::LOG_INFO:
	       .......
	      
I believe our clang settings support this syntax, though I could be wrong

Reply:
I added braces here since using braces consistently for each case enhances code clarity, prevents bugs, and makes code more maintainable. This style aligns with principles of core guidelines which prioritize code readability and maintainability.